src/Adapter/Configuration.php

```diff
@@ -226,10 +228,19 @@ public function combinationIsActive()
 
     /**
      * Restrict updates of a piece of configuration to a single shop.
+     *
      * @param Shop $shop
      */
     public function restrictUpdatesTo(Shop $shop)
     {
         $this->shop = $shop;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBool($key, $default = false)
+    {
+        return (bool) $this->getBoolean($key, $default);
```

Comment:
Useless function (and useless cast in this function). you already have a getBoolean method in ParameterBag :)

Reply:
Agree with useless casting. :+1: 

However, not sure about naming, now in `AdvancedConfigurationInterface` we have `getInt()` and `getBool()`,  if we change it to `getBoolean()` why not changing then `getInt()` to `getInteger()`?

On the other hand, do we really have to use `ParameterBag` at all? For me, it does not look like its a good fit for configuration service. If you take a look, quite a few methods dont even work here (it starts with `all()` implementation or weird constructor implementation).

Is there a use case when you need to get all configuration? Or all configuration keys? Or get count of all configuration? As we can see, currently there isn't, because if there were, code would break. :/

Reply:
Keeping the same way as ParameterBag, we will do less work. More, they use `getInt` and `getBoolean` , don't know why, but it maybe will easier for contributor to have the same way as Symfony does :)

Reply:
@PierreRambaud this is because you assume we're doing a software for Symfony developers, we're not :)

I suggest to keep the full name every time:

* getBoolean()
* getInteger()
* getFloat()
* ...

Reply:
But we are doing a software with Symfony, easier for everyone when data are consistent :/
One time it's getInt, another it's getInteger, getBool, getBoolean, can break some brains :O